studio/tartifact_store.py

```diff
@@ -274,7 +281,10 @@ def finish_download():
                     self.logger.info('tar stdout output: \n ' + str(tarout))
                     self.logger.info('tar stderr output: \n ' + str(tarerr))
 
-                if len(listtar) == 1:
+                if len(listtar) == 1 and not isTarFromDotDir:
+                    # Here we protect ourselves from the corner case,
+                    # when we try to move A/. folder to A.
+                    # os.rename() will fail to do that.
```

Comment:
That's an actual hacky fix for one of the bugs:
in case of empty tar file, we were trying
to move directories A/. to A, which makes OS library very unhappy.